core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala

```diff
@@ -476,3 +480,19 @@ private class ListenerThatAcceptsSparkConf(conf: SparkConf) extends SparkListene
   var count = 0
   override def onJobEnd(job: SparkListenerJobEnd): Unit = count += 1
 }
+
+private class SlowShuffleManager(conf: SparkConf) extends SortShuffleManager(conf) {
+  
+  override val shuffleBlockResolver: IndexShuffleBlockResolver =
+    new SlowIndexShuffleBlockResolver(conf)
+    
+    private class SlowIndexShuffleBlockResolver(conf: SparkConf,
+                                                _blockManager: BlockManager = null)
+      extends IndexShuffleBlockResolver(conf, _blockManager) {
+
+      override def getBlockData(blockId: ShuffleBlockId): ManagedBuffer = {
+        Thread.sleep(1)
+        super.getBlockData(blockId)
+      }
+    }
+}
```

Comment:
I forget whether this will fail style checks without a newline at the end.

Content-wise, this looks good though